auslib/dockerflow.py

```diff
@@ -26,8 +26,12 @@ def heartbeat():
         """Per the Dockerflow spec:
         Respond to /__heartbeat__ with a HTTP 200 or 5xx on error. This should
         depend on services like the database to also ensure they are healthy."""
-        database_entry_value = heartbeat_database_fn(dbo)
-        return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
+        try:
+            database_entry_value = heartbeat_database_fn(dbo)
+            return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
+        except Exception as e:
+            e.status_code = 500
+            return Response(e)
```

Comment:
First of all, please remember that you're supposed to be returning a 502, _not_ a 500.

The reason this isn't working is beacuse you're not constructing a response correctly. Have a look at http://flask.pocoo.org/docs/0.10/api/#flask.Response to see the API of that class. You can also see some examples of setting a custom status code and body with Response if you look around the Balrog code base.